sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_load.hpp

```diff
@@ -445,3 +445,415 @@ template <typename T, bool TestPVCFeatures> bool testACC(queue Q) {
+            << ",UseMask=" << UseMask
+            << ",UsePassThruOperand=" << UsePassThruOperand;
+
+  sycl::range<1> GlobalRange{Groups};
+  sycl::range<1> LocalRange{GroupSize};
+  sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
+
+  constexpr size_t Alignment = getAlignment<T, N, UseMask>(LoadProperties);
+  shared_vector In(Size, shared_allocator{Q});
+  shared_vector Out(Size, shared_allocator{Q});
+  for (int i = 0; i < Size; i++) {
+    In[i] = esimd_test::getRandomValue<T>();
+    Out[i] = 0;
+  }
+
+  try {
+    Q.submit([&](handler &CGH) {
+       local_accessor<T, 1> LocalAcc(GroupSize * N, CGH);
+       auto InPtr = In.data();
+       auto OutPtr = Out.data();
+
+       CGH.parallel_for(Range, [=](sycl::nd_item<1> NDI) SYCL_ESIMD_KERNEL {
+         uint16_t GlobalID = NDI.get_global_id(0);
+         uint16_t LocalID = NDI.get_local_id(0);
+         uint32_t GroupID = NDI.get_group(0);
+         uint32_t LocalElemOffset = LocalID * N;
+         uint32_t GlobalElemOffset = GlobalID * N;
+         if (LocalID == 0)
+           for (int I = 0; I < GroupSize * N; I++)
+             LocalAcc[I] = InPtr[GlobalID * N + I];
+         barrier();
+
+         simd<T, N> Vals;
+         simd_mask<1> Mask = (GlobalID + 1) & 0x1;
+         if constexpr (!CheckProperties) {
+           if constexpr (UsePassThruOperand) {
+             // TODO: these 2 lines work-around the problem with scalar
+             // conversions to bfloat16. It could be just: "simd<T, N>
+             // PassThru(GlobalElemOffset, 1);"
+             simd<uint32_t, N> PassThruInt(GlobalElemOffset, 1);
+             simd<T, N> PassThru = PassThruInt;
+
+             if (LocalElemOffset == 0) // try the variant without byte-offset
+               Vals = block_load<T, N>(LocalAcc, Mask, PassThru);
+             else
+               Vals = block_load<T, N>(LocalAcc, LocalElemOffset * sizeof(T),
+                                       Mask, PassThru);
+           } else { // if !UsePassThruOperand
+             if constexpr (UseMask) {
+               if (LocalElemOffset == 0)
+                 Vals = block_load<T, N>(LocalAcc, Mask);
+               else
+                 Vals = block_load<T, N>(LocalAcc, LocalElemOffset * sizeof(T),
+                                         Mask);
+             } else { // if !UseMask
+               if (LocalElemOffset == 0)
+                 Vals = block_load<T, N>(LocalAcc);
+               else
+                 Vals = block_load<T, N>(LocalAcc, LocalElemOffset * sizeof(T));
+             }
+           }
+         } else { // if CheckProperties
+           if constexpr (UsePassThruOperand) {
+             // TODO: these 2 lines work-around the problem with scalar
+             // conversions to bfloat16. It could be just: "simd<T, N>
+             // PassThru(GlobalElemOffset, 1);"
+             simd<uint32_t, N> PassThruInt(GlobalElemOffset, 1);
+             simd<T, N> PassThru = PassThruInt;
```

Comment:
We could move this outside the ifs so it's not duplicated

Reply:
Fixed in d6202ffa36ff717913cda09b85bcb810b0660de7